cvxpy/reductions/solvers/conic_solvers/scipy_conif.py

```
@@ -0,0 +1,142 @@
+        data[s.PARAM_PROB] = problem
+        data[self.DIMS] = problem.cone_dims
+        inv_data[self.DIMS] = problem.cone_dims
+
+        constr_map = problem.constr_map
+        inv_data[self.EQ_CONSTR] = constr_map[Zero]
+        inv_data[self.NEQ_CONSTR] = constr_map[NonNeg]
+        len_eq = problem.cone_dims.zero
+
+        c, d, A, b = problem.apply_parameters()
+        data[s.C] = c
+        inv_data[s.OFFSET] = d
+        data[s.A] = -A[:len_eq]
+        if data[s.A].shape[0] == 0:
+            data[s.A] = None
+        data[s.B] = b[:len_eq].flatten()
+        if data[s.B].shape[0] == 0:
+            data[s.B] = None
+        data[s.G] = -A[len_eq:]
+        if 0 in data[s.G].shape:
+            data[s.G] = None
+        data[s.H] = b[len_eq:].flatten()
+        if 0 in data[s.H].shape:
+            data[s.H] = None
+        return data, inv_data
+
+    def solve_via_data(self, data, warm_start: bool, verbose: bool, solver_opts, solver_cache=None):
+        from scipy import optimize as opt
+
+        # Check to see if scipy version larger than 1.6.2 is installed
+        import scipy
+        if StrictVersion(scipy.__version__) < StrictVersion('1.6.1'):
+            raise ValueError("This solver requires a SciPy version >= 1.6.1")
+
+        # Extract solver options which are not part of the options dictionary
+        if solver_opts:
+            if "scipy_options" not in solver_opts:
+                raise ValueError("All parameters for the SCIPY solver should "
+                                 "be incased within a dictionary called "
+                                 "scipy_options e.g. \n"
+                                 "prob.solve(solver='SCIPY', verbose=True,"
+                                 " scipy_options={'method':'highs-ds', 'maxiter':10000})")
+            meth = solver_opts['scipy_options'].pop("method", "highs-ds")
+            bnds = solver_opts['scipy_options'].pop("bounds", (None, None))
```

"Bounds" is interpreted with respect to the single vectorized decision variable, which won't be transparently related to the CVXPY model specified by the user. For the time being we should raise an error if a user specifies "bounds".

---

Two of the lp tests (lp_3 and lp_4 in cvxpy/tests/solver_test_helpers.py) were passed and have bounds within the constraints similar to:

`# Infeasible
x = cp.Variable((10, 1))
objective = cp.Minimize(cp.sum(x))
prob = cp.Problem(objective, [x >= 1, x <= 0])
prob.solve(verbose=True, solver="SCIPY")`

As far as I can tell, as long as the bounds for CVXPY are contained within the constraints of the Problem instance, they will be passed to the solver through the constraints. With the "bounds" parameter within the scipy_options dictionary the user has the ability to specify exactly the same bounds:

`# Infeasible
x = cp.Variable((10, 1))
objective = cp.Minimize(cp.sum(x))
prob = cp.Problem(objective)
prob.solve(verbose=True, solver="SCIPY", scipy_options={'bounds':(1, 0)})`

From what I can see, the only downside of keeping the bounds parameter, is that if the user were to try another solver and only specified the bounds through that parameter, then the other solver would not see those bounds. But if the user has specified the bounds using the first example above, then everything should work as normal. 

Should I still go ahead and raise an error if the user specifies "bounds" within scipy_options to force the user to include them within the constraints of the CVXPY Problem instance?

---

The equivalence between specifying bounds in a CVXPY Constraint (or a Variable attribute) and in ``scipy_options`` is only really valid for a problem with a single vectorized decision variable, and for problems that use only linear algebraic constraints. Some of our "atoms" are LP representable but require introducing additional helper variables that the user isn't told about. For example, we introduce helper variables for the 1-norm, infinity-norm, and the sum-k-largest atoms. In these cases, if someone set ``bounds=(1, 10)``, then they would unknowingly affect the helper variables we introduced while canonicalizing the problem. I think the best thing to do right now is to raise an error. @SteveDiamond, @akshayka, what do you two think?

---

I agree that it's best to raise an error. It would be very confusing to have to separate but slightly different notions of box constraints (CVXPY constraints and SciPy bounds).

---

Thanks @akshayka and @rileyjmurray, I have gone ahead and ensured that CVXPY raises an error if the user specifies the 'bounds' parameter.